app/src/editor/shared/components/SearchPanel/index.jsx

```diff
@@ -29,264 +28,266 @@ function MaybeDraggable({ disabled, children, ...props }) {
-        }))
-    }
+    useLayoutEffect(() => {
+        if (layout.resizing) { return } // do nothing while resizing
 
-    onKeyDown = (event) => {
-        if (this.props.isOpen && event.key === 'Escape' && this.state.hasFocus) {
-            this.props.open(false)
-        }
-    }
+        const { current: currentContent } = contentRef
+        const itemsHeight = currentContent ? currentContent.offsetHeight : minHeight
+        const requiredHeight = minHeight + (itemsHeight ? scrollPadding + itemsHeight : 0)
 
-    onInputRef = (el) => {
-        this.input = el
-        setTimeout(() => { // temporary workaround for modal timing
-            if (this.props.isOpen && this.input) {
-                this.input.focus()
+        setLayoutState((layout) => {
+            const { preferredHeight } = layout
+            let { height } = layout
+            if (!isSearching) {
+                // when no search show at either min or preferredHeight height
+                height = isExpanded ? layout.preferredHeight : minHeight
+            } else {
+                // autosize search
+                height = Math.min(Math.max(minHeight, requiredHeight), preferredHeight)
             }
-        }, 100)
-    }
 
-    onInputFocus = () => {
-        if (this.input) {
-            this.input.select()
-        }
-        this.setState({ hasFocus: true })
-    }
-
-    onInputBlur = () => {
-        this.setState({ hasFocus: false })
-    }
+            if (height === layout.height) {
+                return layout // noop if same
+            }
 
-    render() {
-        const {
-            open,
-            isOpen,
-            bounds,
-            placeholder,
-            minWidth,
-            minHeightMinimized,
-            maxWidth,
-            maxHeight,
-            children,
-            className,
-            scrollPadding,
-            dragDisabled,
-            headerHidden,
-        } = this.props
-        const {
-            search,
-            isExpanded,
-            width,
-            posX,
-            posY,
-        } = this.state
-        const height = this.calculateHeight()
-        const isSearching = !!search.trim()
+            return {
+                ...layout,
+                height,
+            }
+        })
+    }, [layout, contentRef, setLayoutState, children, isSearching, minHeight, scrollPadding, isExpanded])
 
-        return (
-            <React.Fragment>
-                <MaybeDraggable
-                    disabled={dragDisabled}
-                    handle={`.${styles.dragHandle}`}
-                    bounds={bounds}
-                    position={{
-                        x: posX,
-                        y: posY,
-                    }}
-                    onStop={(e, data) => this.setState({
-                        posX: data.x,
-                        posY: data.y,
+    return (
+        <React.Fragment>
+            <MaybeDraggable
```

Comment:
I wonder if we could use `Draggable`'s [`axis="none"`](https://github.com/mzabriskie/react-draggable#draggable-props) to disable d&d. Maybe we don't need another component.

Reply:
Good thought, though currently the components that need this disabled keep it disabled for their entire lifetime, so saves the Draggable component from even instantiating if it's not going to be used.